tests/integration/test_pr_agent_config_validation.py

```
@@ -0,0 +1,267 @@
+                else:
+                    pytest.fail(f"YAML parsing error in config: {e}")
+
+    def test_non_hashable_keys_detected(self):
+        """Verify non-hashable keys are detected and raise appropriate errors."""
+        import tempfile
+        import os
+
+        class NonHashableKeyLoader(yaml.SafeLoader):
+            pass
+
+        def construct_mapping_check_hashable(loader, node, deep=False):
```

Ruff (0.14.7)

Missing return type annotation for private function `construct_mapping_check_hashable`

(ANN202)